crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,292 @@
+            })
+            .collect();
+
+        for handle in handles {
+            handle.join().unwrap();
+        }
+
+        let guard = result.lock().unwrap();
+        if let Some(comment) = *guard {
+            return Some(comment);
+        }
+
+        if is_high_entropy(&text) {
+            Some("The string has a high entropy value")
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! { "Potential secret found." },
+            )
+            .note(markup! { "Type of secret detected: " {state} })
+            .note(markup! {
+                "Storing secrets in source code is a security risk. Consider the following steps:"
+                "\n1. Remove the secret from your code."
+                "\n2. If needed, use environment variables or a secure secret management system to store sensitive data."
+                "\n3. If this is a false positive, consider adjusting the rule configuration or adding an inline disable comment."
+            })
+        )
+    }
+}
+
+// Workaround: Since I couldn't figure out how to declare them inline,
+// declare the LazyLock patterns separately
+static SLACK_TOKEN_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"xox[p|b|o|a]-[0-9]{12}-[0-9]{12}-[0-9]{12}-[a-z0-9]{32}").unwrap()
+});
+
+static GENERIC_SECRET_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#"[sS][eE][cC][rR][eE][tT].*[0-9a-zA-Z]{32,45}"#).unwrap());
+
+static GENERIC_API_KEY_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#"[aA][pP][iI][_]?[kK][eE][yY].*[0-9a-zA-Z]{32,45}"#).unwrap());
+
+static SLACK_WEBHOOK_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(
+        r"https://hooks\.slack\.com/services/T[a-zA-Z0-9_]{8}/B[a-zA-Z0-9_]{8}/[a-zA-Z0-9_]{24}",
+    )
+    .unwrap()
+});
+
+static GITHUB_TOKEN_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#"[gG][iI][tT][hH][uU][bB].*[0-9a-zA-Z]{35,40}"#).unwrap());
+
+static TWITTER_OAUTH_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#"[tT][wW][iI][tT][tT][eE][rR].*[0-9a-zA-Z]{35,44}"#).unwrap());
+
+static FACEBOOK_OAUTH_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#"[fF][aA][cC][eE][bB][oO][oO][kK].*[0-9a-f]{32}"#).unwrap());
+
+static HEROKU_API_KEY_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(
+        r"[hH][eE][rR][oO][kK][uU].*[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}",
+    )
+    .unwrap()
+});
+
+static PASSWORD_IN_URL_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r#"[a-zA-Z]{3,10}://[^/\s:@]{3,20}:[^/\s:@]{3,20}@.{1,100}['"\s]"#).unwrap()
+});
+
+static GOOGLE_SERVICE_ACCOUNT_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r#""type": "service_account""#).unwrap());
```

Comment:
This looks like it doesn't need to be a regex.

Reply:
Updated to use a better variant to detect single and double quotes.